core/common/src/main/java/alluxio/grpc/GrpcManagedChannelPool.java

```diff
@@ -222,10 +222,9 @@ public void releaseManagedChannel(ChannelKey channelKey, long shutdownTimeoutMs)
    */
   private ManagedChannel createManagedChannel(ChannelKey channelKey) {
     NettyChannelBuilder channelBuilder;
-    // Use constructor that resolves address if given an unresolved inet address.
-    if (channelKey.mAddress instanceof InetSocketAddress
-        && ((InetSocketAddress) channelKey.mAddress).isUnresolved()) {
+    if (channelKey.mAddress instanceof InetSocketAddress) {
       InetSocketAddress inetServerAddress = (InetSocketAddress) channelKey.mAddress;
+      // This constructor delays DNS lookup to detect changes
```

Comment:
Should we only use this if the address is unresolved?

Reply:
The `InetSocketAddress` passed to grpc should use unresolved hostnmae, otherwise it could get stuck with a stale resolved address.